packages-common/pluggable-widgets-tools/tests/commands.js

```diff
@@ -0,0 +1,175 @@
+const { Mutex } = require("async-mutex");
+const { exec } = require("child_process");
+const fkill = require("fkill");
+const { copy, readJson, writeJson } = require("fs-extra");
+const { join } = require("path");
+const { mkdir, rm, tempdir } = require("shelljs");
+const { promisify } = require("util");
+const { run: runYeoman } = require("yeoman-test");
+
+const LIMIT_TESTS = process.argv.includes("--limited");
+
+const CONFIGS = [
+    ["web", "ts", "8.0"],
+    ["native", "js", "8.1"],
+    ["web", "js", "8.3"],
+    ["native", "ts", "8.6"],
+    ["web", "ts", "8.6"],
+    ["web", "js", "8.7"],
+    ["web", "ts", "latest"],
+    ["native", "js", "latest"]
+];
+
+if (LIMIT_TESTS) {
+    CONFIGS.splice(1, CONFIGS.length - 2); // Remove all configs except the first and the last
+}
+
+const mutex = new Mutex();
+
+main().catch(e => {
+    console.error(e);
+    process.exit(1);
+});
+
+async function main() {
+    console.log("Preparing...");
+
+    const { stdout: packOutput } = await execAsync("npm pack --loglevel=error", join(__dirname, ".."));
+    const toolsPackagePath = join(__dirname, "..", packOutput.trim());
+
+    const failures = (
+        await Promise.all(
+            CONFIGS.map(async config => {
+                try {
+                    await runTest(...config);
+                    return undefined;
+                } catch (e) {
+                    return [config, e];
+                }
+            })
+        )
+    ).filter(f => f);
+
+    rm(toolsPackagePath);
+
+    if (failures.length) {
+        failures.forEach(f => console.error(`Test for configuration ${f[0]} failed: ${f[1]}`));
+        process.exit(2);
+    }
+
+    async function runTest(platform, lang, version) {
+        const widgetName = `generated_${version.replace(".", "_")}_${lang}_${platform}`;
+        const workDir = join(tempdir(), `pwt_test_${Math.round(Math.random() * 10000)}`);
+
+        try {
+            console.log(`[${widgetName}] Preparing widget...`);
+            await prepareWidget();
+            console.log(`[${widgetName}] Ready to test!`);
+
+            console.log(`[${widgetName}] Testing linting...`);
+            await testLint();
+            console.log(`[${widgetName}] Testing unit tests....`);
+            if (platform === "native") {
+                await testTest();
+            } else {
+                await execAsync("npm run test", workDir);
+            }
+
+            if (LIMIT_TESTS) {
+                console.log(`[${widgetName}] Quick tested!`);
+                return;
+            }
+
+            for (const cmd of ["build", "test:unit", "release"]) {
```

Comment:
You are testing unit tests twice.

Reply:
Surprisingly, but `test:unit` is actually doing something slightly different than `test` covered above.

Reply:
I see, `test:unit` does coverage. `test` doesn't.